lib/wallet/src/Cardano/Wallet/IO/Delegation.hs

```diff
@@ -122,13 +122,18 @@ selectCoinsForJoin ctx pools poolId poolStatus = do
         <- W.readNodeTipStateForTxWrite netLayer
     currentEpochSlotting <- W.getCurrentEpochSlotting netLayer
 
-    action <- WD.joinStakePoolDelegationAction @s
+    inConway <- case W.votingEnabledInEra era of
+        Left _ -> pure False
+        Right _ -> pure True
+
+    (action,_) <- WD.joinStakePoolDelegationAction @s
```

Comment:
🤔 Shouldn't the voting certificate also be part of the transaction that we select coins for?

Reply:
@Anviking @jonathanknowles   ????

Reply:
So this endpoint is used by Daedalus for hardware wallet support.

If we want voting supported there, we should ensure the certificate is accounted for in the coin-selection, yes, but also that it is returned in the response. And then Daedalus would need to be updated to add that certificate to the tx.

Reply:
> And then Daedalus would need to be updated to add that certificate to the tx.

Dang. We can select the coins for it, but we don't assemble the transaction — it looks like Daedalus can't even vote to abstain in the presence of a hardware wallet.